kyaml/yaml/walk/associative_sequence.go

```diff
@@ -55,12 +55,18 @@ func appendListNode(dst, src *yaml.RNode, keys []string) (*yaml.RNode, error) {
 
 		// We use the key and value from elem to find the corresponding element in dst.
 		// Then we will use ElementSetter to replace the element with elem. If we cannot
```

Comment:
The comment above seems cut off.

Reply:
It's great that this PR adds a krusty test in the kustomize api module.

But there's a dearth of in-module testing here.  This code is too fundamentally important not to have direct in-module coverage.  

There's  no `_test.go` file in the kyaml module that calls Walker directly.  But there's a good framework in `merge2_test.go` that indirectly hits walker.  Can a test covering this port issue be added there?

Reply:
yup, makes sense to me, I'll add a couple of tests to merge2 right now